internal/vfs/vfstest/vfstest.go

```diff
@@ -482,20 +523,54 @@ func (m *mapFS) WriteFile(path string, data []byte, perm fs.FileMode) error {
 
 	m.setEntry(path, cp, fstest.MapFile{
 		Data:    data,
-		ModTime: time.Now(),
+		ModTime: m.clock.Now(),
 		Mode:    perm &^ umask,
 	})
 
 	return nil
 }
 
-func (m *mapFS) Remove(path string) error {
+func (m *MapFS) Remove(path string) error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
 	return m.remove(path)
 }
 
+func (m *MapFS) GetTargetOfSymlink(path string) (string, bool) {
+	path, _ = strings.CutPrefix(path, "/")
+	m.mu.RLock()
+	defer m.mu.RUnlock()
+	canonical := m.getCanonicalPath(path)
+	canonicalString := string(canonical)
+	if fileInfo, ok := m.m[canonicalString]; ok {
+		if fileInfo.Mode&fs.ModeSymlink != 0 {
+			return "/" + string(fileInfo.Data), true
+		}
+	}
+	return "", false
+}
+
+func (m *MapFS) Entries() iter.Seq2[string, *fstest.MapFile] {
+	return func(yield func(string, *fstest.MapFile) bool) {
+		m.mu.RLock()
+		defer m.mu.RUnlock()
```

Comment:
I'm not sure if it much matters, but this will effectively mean that the whole MapFS is locked until iteration completes.